src/tests/unit/auto/get_device_list.cpp

```diff
@@ -79,40 +88,63 @@ class GetDeviceListTest : public ::testing::TestWithParam<ConfigParams> {
     EXPECT_CALL(*core, GetAvailableDevices()).Times(1);
     auto result = plugin->GetDeviceList({{ov::device::priorities.name(), priorityDevices}});
     EXPECT_EQ(result, metaDevices);
 }
 
-
-// ConfigParams details
-// example
-// ConfigParams {devicePriority, expect metaDevices, ifThrowException}
-
-const std::vector<ConfigParams> testConfigs = {
-    //
-    ConfigParams {"CPU,GPU,VPUX",
-        "CPU,GPU,VPUX"},
-    ConfigParams {"VPUX,GPU,CPU,-GPU.0",
-        "VPUX,GPU.1,CPU"},
-    ConfigParams {"-GPU.0,GPU,CPU",
-        "GPU.1,CPU"},
-    ConfigParams {"-GPU.0,GPU",
-        "GPU.1"},
-    ConfigParams {"-GPU.0", "CPU,GPU.1,VPUX"},
-    ConfigParams {"-GPU.0,-GPU.1", "CPU,VPUX"},
-    ConfigParams {"-GPU.0,-CPU", "GPU.1,VPUX"}
-};
-
-
-INSTANTIATE_TEST_SUITE_P(smoke_Auto_BehaviorTests, GetDeviceListTest,
-                ::testing::ValuesIn(testConfigs),
-            GetDeviceListTest::getTestCaseName);
+const std::vector<Params> testConfigsWithId = {Params{" ", " "},
+                                         Params{"", "CPU,GPU.0,GPU.1"},
+                                         Params{"CPU, ", "CPU, "},
+                                         Params{" ,CPU", " ,CPU"},
+                                         Params{"CPU,", "CPU"},
+                                         Params{"CPU,,GPU", "CPU,GPU.0,GPU.1"},
+                                         Params{"CPU, ,GPU", "CPU, ,GPU.0,GPU.1"},
+                                         Params{"CPU,GPU,VPUX,INVALID_DEVICE", "CPU,GPU.0,GPU.1,VPUX,INVALID_DEVICE"},
+                                         Params{"VPUX,GPU,CPU,-GPU.0", "VPUX,GPU.1,CPU"},
+                                         Params{"-GPU.0,GPU,CPU", "GPU.1,CPU"},
+                                         Params{"-GPU.0,GPU", "GPU.1"},
+                                         Params{"-GPU.0", "CPU,GPU.1"},
+                                         Params{"-GPU.0,-GPU.1", "CPU"},
+                                         Params{"-GPU.0,-GPU.1,INVALID_DEVICE", "INVALID_DEVICE"},
+                                         Params{"-GPU.0,-CPU", "GPU.1"}};
+
+const std::vector<Params> testConfigs = {Params{" ", " "},
+                                         Params{"", "CPU,GPU"},
+                                         Params{"GPU", "GPU"},
+                                         Params{"GPU.0", "GPU.0"},
+                                         Params{"GPU,GPU.0", "GPU,GPU.0"},
+                                         Params{"CPU", "CPU"},
+                                         Params{" ,CPU", " ,CPU"},
+                                         Params{" ,GPU", " ,GPU"},
+                                         Params{"GPU, ", "GPU, "},
+                                         Params{"CPU,GPU", "CPU,GPU"},
+                                         Params{"CPU,-GPU", "CPU"},
+                                         Params{"CPU,-GPU,GPU.0", "CPU,GPU.0"},
```

Comment:
`-GPU` mean remove all GPU devices. Why we still have GPU.0 in result?

Reply:
I think It may be reasonable to keep "GPU.0" because user has specified the "GPU.0" in the device priority. "-GPU"  means to remove the rest sub-device of GPU like "GPU.1".